fearless/Modules/Staking/StakingMain/StakingMainViewController.swift

```diff
@@ -409,27 +407,21 @@ extension StakingMainViewController: UICollectionViewDelegate {
     }
 }
 
-// MARK: Nomination View Delegate -
+// MARK: - StakingStateViewDelegate
 
-extension StakingMainViewController: NominationViewDelegate {
-    func nominationViewDidReceiveMoreAction(_: NominationView) {
+extension StakingMainViewController: StakingStateViewDelegate {
+    func stakingStateViewDidReceiveMoreAction(_: StakingStateView) {
         presenter.performManageStakingAction()
     }
 
-    func nominationViewDidReceiveStatusAction(_: NominationView) {
-        presenter.performNominationStatusAction()
-    }
-}
-
-// MARK: - ValidationViewDelegate
-
-extension StakingMainViewController: ValidationViewDelegate {
-    func validationViewDidReceiveMoreAction(_: ValidationView) {
-        presenter.performManageStakingAction()
-    }
+    func stakingStateViewDidReceiveStatusAction(_ view: StakingStateView) {
+        if view is NominatorStateView {
+            presenter.performNominationStatusAction()
+        }
 
-    func validationViewDidReceiveStatusAction(_: ValidationView) {
-        presenter.performValidationStatusAction()
+        if view is ValidatorStateView {
```

Comment:
Can view be both Nominator and Validator state view? If not, I suggest to put `else` or `return` to indicate that